### ISSUE TITLE
Limits in giving commands in R-Instat?

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4414]
We can now give commands to R-Instat in 2 ways:
1) Through the Prepare > Calculations > Calculate dialogue
2) Through the script window

And Maxwell has been adding to the right-click in the script window, being able to save and load, and paste etc.  This is a facility that many people have asked for, and I have found both these ways to be useful.

I need to understand more about the limitations of these facilities.  I recognise there are risks, and would also like to be able to warn users about them too.

This has arisen partly because of the value of giving a command to R-Instat that is not yet available in a dialogue.  Is this possible?  Is it desirable?
Here is an example:

update_geom_defaults("point", list(colour = "darkblue"))

This is a command that would be useful.  It doesn't need to be stored, so doesn't need to add to the R-Instat?

If I can give this command, could I give 2 of them together, e.g. also
update_geom_defaults("point", list(size = "3.5"))
We can now give commands to R-Instat in 2 ways:
1) Through the Prepare > Calculations > Calculate dialogue
2) Through the script window

And Maxwell has been adding to the right-click in the script window, being able to save and load, and paste etc.  This is a facility that many people have asked for, and I have found both these ways to be useful.

I need to understand more about the limitations of these facilities.  I recognise there are risks, and would also like to be able to warn users about them too.

This has arisen partly because of the value of giving a command to R-Instat that is not yet available in a dialogue.  Is this possible?  Is it desirable?
Here is an example:

update_geom_defaults("point", list(colour = "darkblue"))

This is a command that would be useful.  It doesn't need to be stored, so doesn't need to add to the R-Instat?

If I can give this command, could I give 2 of them together, e.g. also
update_geom_defaults("point", list(size = "3.5"))

**Comment by dannyparsons** [comment]
Yes that's completely possible. The only limitation on commands is that the package must be available on your machine. So its powerful but dangerous.

**Comment by rdstern** [comment]
Wow.  Dangerous is my middle name.  I must try some examples and report.

**Comment by dannyparsons** [comment]
Try, for example, a very short command: `q()`

**Comment by rdstern** [comment]
That's not dangerous; just a pretty efficient way of stopping the session!  Just the sort of thing to (perhaps) not put into our help - or lots of people might then try it.
